File: modules/system-sounds/ios/SystemSoundsModule.swift
import ExpoModulesCore

public class SystemSoundsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("SystemSounds")

    AsyncFunction("listSounds") { (_kind: String) -> [[String: Any]] in
      // iOS does not expose the system ringtone catalog to third-party apps.
      return []
    }

    AsyncFunction("listPicks") { (_limit: Int) -> [[String: Any]] in
      return []
    }

    AsyncFunction("play") { (_uri: String, _loop: Bool) in
    }

    AsyncFunction("playResource") { (_name: String, _loop: Bool) in
    }

    AsyncFunction("stop") {
    }

    AsyncFunction("presentAlarmNotification") { (
      _identifier: String,
      _title: String,
      _body: String,
      _alarmId: String,
      _channelId: String,
      _launchUrl: String,
      _silent: Bool
    ) in
    }

    AsyncFunction("dismissAlarmNotification") { (_identifier: String) in
    }

    AsyncFunction("acquireWakeLock") {
    }

    AsyncFunction("releaseWakeLock") {
    }

    AsyncFunction("canUseFullScreenIntent") { () -> Bool in
      return false
    }

    AsyncFunction("openFullScreenIntentSettings") {
    }
  }
}

File: modules/system-sounds/src/SystemSoundsModule.ts
import { NativeModule, requireNativeModule } from "expo";

import type { SystemSoundItem } from "./SystemSounds.types";

declare class SystemSoundsModule extends NativeModule<Record<string, never>> {
  listSounds(kind: string): Promise<SystemSoundItem[]>;
  listPicks(limit: number): Promise<SystemSoundItem[]>;
  play(uri: string, loop: boolean): Promise<void>;
  playResource(name: string, loop: boolean): Promise<void>;
  stop(): Promise<void>;
  presentAlarmNotification(
    identifier: string,
    title: string,
    body: string,
    alarmId: string,
    channelId: string,
    launchUrl: string,
    silent: boolean,
  ): Promise<void>;
  dismissAlarmNotification(identifier: string): Promise<void>;
  acquireWakeLock(): Promise<void>;
  releaseWakeLock(): Promise<void>;
  canUseFullScreenIntent(): Promise<boolean>;
  openFullScreenIntentSettings(): Promise<void>;
}

export default requireNativeModule<SystemSoundsModule>("SystemSounds");
